Makefile

```diff
@@ -36,11 +36,41 @@ CPPFLAGS += -DHISTFILE=\"$(HISTFILE)\" \
 OBJS += linenoise-1.0/linenoise.o
 linenoise-1.0/linenoise.o : linenoise-1.0/linenoise.c linenoise-1.0/linenoise.h
 
-## Comment these IN to enable custom baudrate support.
-## Currently works *only* with Linux (kernels > 2.6).
-#CPPFLAGS += -DUSE_CUSTOM_BAUD
-#OBJS += termios2.o
-#termios2.o : termios2.c termios2.h termbits2.h
+## Detect host machine / OS and store it into a make variable named "HOST_SYSTEM" for later use
```

Comment:
I don't like this. Please keep the Makefile simple (not too many GNU-Makeisms). I want it to be able to work with other make's as well. It's better to have the user comment-out some lines rather than auto-detect things. Put instead something like this for MAC OSX:

    # coment this in to enable custom baudrate support for OSX
    #CPPFLAGS += -DUSE_CUSTOM_BAUD

    # Comment this in to enable custom baudrate support for Linux
    # (Kernels > 2.6)
    #CPPFLAGS += -DUSE_CUSTOM_BAUD
    #OBJS += ...